scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,141 @@
+  """
+    col_map_obj = GenerateColMapBase(specDict=specDict, columnList=columnList)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+  """
+
+    def __init__(self, specDict=None, columnList=None):
+        """module init """
+        self.features = specDict
+        self.column_list = columnList
+        self.column_map = {}
+
+    def _generate_stat_vars_from_spec(self):
+        """generates stat_var nodes for each column in column list and returns a dictionary of columns with their respective stat_var nodes"""
+        # for each column generate the definition of their respective statistical variable node
+        for col in self.column_list:
+            if 'ignoreColumns' in self.features and col not in self.features[
+                    'ignoreColumns']:
+                self.column_map[col] = self._column_to_statvar(col)
+        #TODO: Enable validating the generate columnMap: _valid_col_map():
+        return self.column_map
+
+    def _column_to_statvar(self, column):
+        """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+        measurement_assigned = False
+        stat_var = {}
+        # part_list contains a list of tokens of the column name after splitting the
+        # column name based on the '!!' delimiter
+        part_list = column.split('!!')
+
+        # part_list[-1], the last element of part_list describes the property value.
+        # This element points to the 'v' in 'pv' -> and we map it to the
+        # corresponding property 'p'
+        # Eg:
+        # 'Estimate!!Total!!Total civilian noninstitutionalized
+        # population!!SEX!!Male' is the name of the column and the value you we want
+        # is the gender 'Male'. This pattern repeats for all columns
+        # To verify this assumption for specific census tables, please check the
+        # table shells at: https://www2.census.gov/programs-surveys/acs/tech_docs/table_shells/2019/
+        val_prop_col = part_list[-1]
+
+        for part in part_list:
+            # set the base for special cases like median, etc.
+            if not measurement_assigned and 'measurement' in self.features.keys(
+            ):
+                if part in self.features['measurement'].keys():
+                    stat_var.update(self.features['measurement'][part])
+                    measurement_assigned = True
+
+        # set the default statVar definition
+        if not measurement_assigned and 'measurement' in self.features.keys(
```

Comment:
.keys() is not required to check if a key is present:
if 'measurement' in self.features: